### DISCUSSION TITLE
No automatic Reconnect to Libre3

### DISCUSSION THREAD

**Discussion by kraftm1989** [j-kaltes/Juggluco#7] [Q&A]
Hello together!
I have the problem that a connection to the Libre3 sensor is not re-established automatically if it has been unreachable for a few minutes (maximum 10).
I have to re-scan it via NFC to get everything working again.
I use Android 12, have already checked the battery settings and permissions and tried the "Wakelock" and "Turbo" options in Juggluco.
With the patched Libre app for the Libre2 I did not know such problems so far (on an otherwise unchanged phone).
Is there a solution for this?
Greetings
Max

**Comment by j-kaltes** [comment]
How long did you wait after getting in to reach again before taking these actions?
Can you send me a screenshot of left menu->sensor next time you don't receive glucose values within 10 minutes after getting into reach again, without first taking any action?
Turning Bluetooth off and on is the more usual action when the Bluetooth connection somehow freezes.

**Comment by kraftm1989** [comment]
After losing the connection I've been sitting dectly next to my mobile for at least ten minutes...without reconnect.

![Screenshot_2023-03-05-21-12-36-90_f82e91c92ccf8465af83f3368bd404b3](https://user-images.githubusercontent.com/41928733/222988780-4384a3d9-4a83-4ebf-84be-ccd1318f7566.jpg)

After the screenshot it turned Bluetooth off an on again - and it reconnected.

**Reply by j-kaltes** [reply]
Juggluco asks Android for a connection after a disconnection. It is the same code as used with Freestyle Libre 2 and that does always work.
Did you disable all battery optimizations for Juggluco?
What phone do you have?

**Comment by kraftm1989** [comment]
Afaik anything is deactivated for juggluco.
![Screenshot_2023-03-06-09-38-52-66_0ba066473b79d6e213a1f6f52505e2ee](https://user-images.githubusercontent.com/41928733/223060585-f10b4521-ae4d-4f5c-93c7-517e71037b58.jpg)

Its an oneplus Nord 2

**Comment by j-kaltes** [comment]
Did you look at https://dontkillmyapp.com/oneplus  and https://nine-faq.9folders.com/articles/11588-how-to-turn-off-battery-optimization-on-the-oneplus? There seem to be some additional battery optimizations.

**Reply by kraftm1989** [reply]
Yes, I know the recommendations. They differ a bit depending on the OS version and mobile phone, but I'm very sure I set the same parameters as for AAPS, xDrip+ and before that the Libre2 app. I have now also checked this several times.
I have now only deactivated a "standby optimisation during sleep" - although I also have the problem during the day.
As I said, I did not experience such problems with the patched Libre2 app. With the config otherwise unchanged.

Translated with www.DeepL.com/Translator (free version)

**Comment by j-kaltes** [comment]
On all phones I have tested, I have never seen that in the sensor screen no recent time was shown. Always there is a connection error (or some other error. )
All in line with the fact that BluetoothDevice.connectGatt is called from Libre3GattCallback.onConnectionStateChange when there is a disconnection error.
